### PR TITLE
Initial Azure Pipelines build configuration

### PR DESCRIPTION
**Pull request by sharwell**
This pull request moves the testing portion of PRs to Azure Pipelines for improved performance. AppVeyor builds are still active since Azure Pipelines does not provide the artifact support required by our status page.

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
+    <clear />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
```
Comment (jnm2) [inline NuGet.config]: Why might I want to consider doing this?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  demands:
+  - msbuild
+  - visualstudio
+  - vstest
+
+jobs:
+- job: Build
+  variables:
+    BuildSolution: StyleCopAnalyzers.sln
+    BuildPlatform: Any CPU
```
Comment (jnm2) [inline azure-pipelines.yml]: I usually remove this variable.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    displayName: 'Use NuGet 5.3.1'
+    inputs:
+      versionSpec: 5.3.1
+
+  - task: NuGetCommand@2
+    displayName: 'NuGet restore'
+    inputs:
+      restoreSolution: '$(BuildSolution)'
+      feedsToUse: 'config'
+      nugetConfigPath: 'NuGet.config'
```
Comment (jnm2) [inline azure-pipelines.yml]: Vastly prefer `msbuildArgs: '/restore'`

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  - task: VSBuild@1
+    displayName: 'Build solution StyleCopAnalyzers.sln'
+    inputs:
+      solution: '$(BuildSolution)'
```
Comment (jnm2) [inline azure-pipelines.yml]: I usually remove this variable too and either remove `solution` or do `solution: .`.

File: azure-pipelines.yml (hunk 3 above)
Comment (sharwell) [reply azure-pipelines.yml]: We have to restore packages from a packages.config file.

File: NuGet.config (hunk 1 above)
Comment (sharwell) [reply NuGet.config]: It's in the commit message https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/3072/commits/a5cbd747cdc7e656f6d1815bbdcb7098b1e0123a

File: azure-pipelines.yml (hunk 3 above)
Comment (jnm2) [reply azure-pipelines.yml]: Whoa, why is that?

File: NuGet.config (hunk 1 above)
Comment (jnm2) [reply NuGet.config]: Perfect, looked there but didn't think of looking for a multiline message. Thanks!

File: azure-pipelines.yml (hunk 3 above)
Comment (sharwell) [reply azure-pipelines.yml]: It's how we bring in the test runner and code coverage tools today.

**Pull request merged**